__tests__/index.test.js

```diff
@@ -0,0 +1,290 @@
+/* eslint-disable jest-formatting/padding-around-test-blocks */
+const nock = require('nock');
+const { join } = require('path');
+const findCacheDir = require('find-cache-dir');
+const fsMock = require('mock-fs');
+const fs = require('fs').promises;
+const api = require('../src');
+const Cache = require('../src/cache');
+const pkg = require('../package.json');
+
+const serverUrl = 'https://api.example.com';
+const createOas = require('./__fixtures__/createOas')(serverUrl);
+
+const examplesDir = join(__dirname, 'examples');
+
+let petstoreSdk;
+let readmeSdk;
+const petstoreServerUrl = 'http://petstore.swagger.io/api';
+
+beforeEach(async () => {
+  fsMock({
+    [examplesDir]: {
+      'petstore.json': await fs.readFile(
+        join(__dirname, '../node_modules/@readme/oas-examples/3.0/json/petstore-expanded.json'),
+        'utf8'
+      ),
+      'readme.json': await fs.readFile(
+        join(__dirname, '../node_modules/@readme/oas-examples/3.0/json/readme.json'),
+        'utf8'
+      ),
+      'uspto.json': await fs.readFile(
+        join(__dirname, '../node_modules/@readme/oas-examples/3.0/json/uspto.json'),
+        'utf8'
+      ),
+    },
+    [findCacheDir({ name: pkg.name })]: {},
+  });
+
+  const petstore = join(examplesDir, 'petstore.json');
+  await new Cache(petstore).saveFile();
+  petstoreSdk = api(petstore);
+
+  const readme = join(examplesDir, 'readme.json');
+  await new Cache(readme).saveFile();
+  readmeSdk = api(readme);
+});
+
+afterEach(() => {
+  fsMock.restore();
+});
+
+describe('#preloading', () => {
+  const uspto = join(examplesDir, 'uspto.json');
+
+  it('should proxy an sdk for the first time', async () => {
+    const mock = nock('https://developer.uspto.gov/ds-api').get('/').reply(200);
+
+    // Asserting that we have not previously loaded this API.
+    expect(new Cache(uspto).isCached()).toBe(false);
+
+    const sdk = api(uspto);
+
+    // SDK should still not be loaded since we haven't officially called it yet.
+    expect(new Cache(uspto).isCached()).toBe(false);
+    expect(Object.keys(sdk)).toStrictEqual(['auth']);
+
+    await sdk.get('/').then(() => {
+      mock.done();
+    });
+
+    // Now that we've called something on the SDK, it should now be fully loaded.
+    expect(new Cache(uspto).isCached()).toBe(true);
+    expect(Object.keys(sdk)).toStrictEqual([
+      'auth',
+      'get',
+      'put',
+      'post',
+      'delete',
+      'options',
+      'head',
+      'patch',
+      'trace',
+      'list-data-sets',
+      'list-searchable-fields',
+      'perform-search',
+    ]);
+  });
+
+  it.todo('should error if passing in swagger 2');
```

Comment:
Sounds like all of these can be removed now they've been implemented in the cache?

Reply:
Yeah this and a couple other todos can get removed.